src/org/exist/xquery/value/DurationValue.java

```diff
@@ -362,7 +364,7 @@ public boolean effectiveBooleanValue() throws XPathException {
     
     public static boolean areReallyEqual(Duration duration1, Duration duration2) {
     	final boolean secondsEqual = zeroIfNull((BigDecimal)duration1.getField(DatatypeConstants.SECONDS)).compareTo(
```

Comment:
I haven't changed this in my commit. That is out of scope here, feel free to fix in a future PR.